CMakeLists.txt

```diff
@@ -63,9 +63,23 @@ else()
     set(SFML_BUILD_EXAMPLES FALSE)
 endif()
 
+# add options for installing the examples
+if(SFML_BUILD_EXAMPLES)
+    sfml_set_option(SFML_INSTALL_EXAMPLES FALSE BOOL "TRUE to install the SFML examples, FALSE to leave them in the build directory")
+    sfml_set_option(SFML_INSTALL_EXAMPLES_PREFIX ${INSTALL_MISC_DIR} PATH "Install path prefix, prepended onto example install directory")
+endif()
+
 # add an option for building the API documentation
 sfml_set_option(SFML_BUILD_DOC FALSE BOOL "TRUE to generate the API documentation, FALSE to ignore it")
 
+# add option for installing the API documentation
+if(SFML_BUILD_DOC)
+    sfml_set_option(SFML_INSTALL_DOC FALSE BOOL "TRUE to install the SFML API documentation, FALSE to leave them in the build directory")
+endif()
+
+# add option for specifying path to install documentation files
+sfml_set_option(SFML_INSTALL_DOC_PREFIX ${INSTALL_MISC_DIR} PATH "Install path prefix, prepended onto the documentation install directory")
```

Comment:
@TankOs argued that "documentation" is not just the API doc, but also the license or something like that.

The problem I saw, that if I use `${INSTALL_MISC_DIR}` like it is currently, you can't really change that, so you'd specify to install the doc at place X and the examples a place Y, but the readme and license file get still installed to the shared/SFML directory or something.

As with the whole issue itself, I don't really care where stuff gets installed, so I'm fine with any path.